lib/APPSensorFusion/App.cpp

```diff
@@ -164,6 +166,19 @@ void App::sendSensorData() const
     payload.magnetometerValueY  = magnetometerValues.valueY;
     payload.turnRate            = turnRates.valueZ;
 
+    uint32_t duration = 0U;
+    if (true == m_firstIteration)
+    {
+        duration         = SEND_SENSOR_DATA_PERIOD;
+        m_firstIteration = false;
+    }
+    duration = m_measurementTimer.getCurrentDuration();
+    m_measurementTimer.restart();
+
+    /* Casting is not problematic since the theoretical time step is much lower than the numerical limits of the used 16
+     * bit unsigned integer. */
+    payload.timePeriod = static_cast<uint16_t>(duration);

```

Comment:
Why dont you define timePeriod as uint32_t directly?

Reply:
The value of timePeriod is around 20. 
SimpleTimer::getCurrentDuration delivers a uint32_t.
I thought it would be unnecessary to transmit at least 2 unnecessary bytes via SerialMuxProt.
However, I will of course change it if you prefer not casting it.